### PR TITLE
SUP-3537 configurable wrapping div with class for table

### PR DESCRIPTION
**Pull request by bernhardriegler**
# added wrapping div element for tables for responsive tables

Create responsive tables by wrapping table-elements created by Aloha-Editor in a div-element with a configurable class to make them scroll horizontally on small devices.
This follows the example giving by "Twitter Bootstrap": http://getbootstrap.com/css/#tables-responsive

## Configuration Example

```
// enable the wrapping div-element and set the class to 'responsive-table'
table.settings.wrapClass = 'responsive-table';

```

## Output Example

```
<div class="responsive-table">
    <table>
        ...
    </table>
</div>
```

### CODE REVIEW

File: build/changelog/entries/2016/11/11291.SUP-3537.feature [hunk 1]
```diff
@@ -0,0 +1,21 @@
+h2. added wrapping div element for tables for responsive tables
+
+Create responsive tables by wrapping table-elements created by Aloha-Editor in a div-element with a configurable class to make them scroll horizontally on small devices.
+This follows the example giving by "Twitter Bootstrap":http://getbootstrap.com/css/#tables-responsive
+
+h3. Configuration Example
+
+<pre>
+// enable the wrapping div-element and set the class to 'responsive-table'
+table.settings.wrapClass = 'responsive-table';
```
Comment (npomaroli) [inline build/changelog/entries/2016/11/11291.SUP-3537.feature]: Aloha.settings.plugins.table.wrapClass = 'responsive-table';

File: src/plugins/common/table/lib/table.js [hunk 2]
```diff
@@ -114,6 +114,17 @@ define([
 		// set the table attribut "obj" as a jquery represenation of the dom-table
 		this.obj = jQuery( table );
 
+		// check config if table should be wrapped
+		if( tablePlugin.settings.wrapClass !== undefined ) {
```
Comment (npomaroli) [inline src/plugins/common/table/lib/table.js]: formatting

File: src/plugins/common/table/lib/table.js [hunk 3]
```diff
@@ -114,6 +114,17 @@ define([
...
+			// add div with class from config
+			this.obj.wrap('<div class="' + tablePlugin.settings.wrapClass + '"></div>');
```
Comment (npomaroli) [inline src/plugins/common/table/lib/table.js]: Check whether wrapping div already exists.
* If exists and classes match, do nothing
* If exists and classes do not match, change classes
* If exists and should not, remove
* If not exists and should, create

File: src/plugins/common/table/lib/table.js [hunk 4]
```diff
@@ -614,14 +625,13 @@ define([
 		// were created dynamically before) ;)
 
 		htmlTableWrapper = this.obj.parents( '.' + this.get( 'classTableWrapper' ) );
-		htmlTableWrapper.get( 0 ).onresizestart = function ( e ) { return false; };
-		htmlTableWrapper.get( 0 ).oncontrolselect = function ( e ) { return false; };
-		htmlTableWrapper.get( 0 ).ondragstart = function ( e ) { return false; };
-		htmlTableWrapper.get( 0 ).onmovestart = function ( e ) { return false; };
+        htmlTableWrapperElem = this.obj.parents( '.' + this.get( 'classTableWrapper' ) ).get( 0 );
```
Comment (npomaroli) [inline src/plugins/common/table/lib/table.js]: convert spaces to tabs

**Comment by bernhardriegler** [comment]
thank you for the feedback - I did add the requested changes

**Comment by npomaroli** [comment]
runbuild

**Pull request merged**